walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,83 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> requiredTimeRanges = findTimeSlotsForNewMeeting(
+        filterRelevantEvents(events, request.getAttendees()),
+        request.getDuration());
```

Comment:
you could simplify your life by casting this `long` to an `int` before everything. We cannot schedule events that are longer than a whole day, so that should help.

Additionally this is a `public` method (think of it as an API), so we could get any kind of foobar here. Testing for corner cases (e.g. evey long events, negative duration, etc) might be a good idea at the start.

Reply:
Done. I'm not sure if I should wrap this duration checks in additional function, for now it's in if.